kore/src/Kore/Syntax/Json.hs

```diff
@@ -124,6 +125,8 @@ fromSubstitution sort subst
             . fromTermLike
             . foldl1 TermLike.mkAnd
             . map toEquals
+            . Set.toList
+            . Set.fromList
```

Comment:
Of course this only addresses cases where the _same_ `Assignment` is duplicated in the substitution.
In `booster` there is code (in the pattern internalisation) which vets sets of assignments to decide what is a true substitution and what isn't.
If we continue to have problem reports we should find out whether and where `kore` generates invalid substitutions